protocols/gossipsub/src/protocol.rs

```diff
@@ -285,17 +287,21 @@ impl Decoder for GossipsubCodec {
     type Error = GossipsubHandlerError;
 
     fn decode(&mut self, src: &mut BytesMut) -> Result<Option<Self::Item>, Self::Error> {
+        // TODO: Replace with codec::decode call
+
         let packet = match self.length_codec.decode(src).map_err(|e| {
             if let std::io::ErrorKind::PermissionDenied = e.kind() {
                 GossipsubHandlerError::MaxTransmissionSize
             } else {
-                GossipsubHandlerError::Io(e)
+                GossipsubHandlerError::Codec(prost_codec::Error::Io(e))
```

Comment:
This is temporary code and will probably be removed once the work on the `decode` begins in earnest.